File: src/main/java/backend/controller/RequestBodies/PostUserBody.java
package backend.controller.RequestBodies;

import org.hibernate.validator.constraints.Email;
import org.hibernate.validator.constraints.NotEmpty;

public class PostUserBody {
    @NotEmpty private String firstname;
    @NotEmpty private String lastname;
    @NotEmpty @Email private String email;
    @NotEmpty private String password;
    @NotEmpty private String gender;


    // --- Getter and Setter ---
    public String getFirstname() {
        return firstname;
    }

    public void setFirstname(String firstname) {
        this.firstname = firstname;
    }

    public String getLastname() {
        return lastname;
    }

    public void setLastname(String lastname) {
        this.lastname = lastname;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }
}

File: src/main/java/backend/controller/RequestBodies/PostUserBody.kt
@file:JvmName("PostUserBody")

package backend.controller.RequestBodies

import org.hibernate.validator.constraints.Email
import org.hibernate.validator.constraints.NotEmpty

class PostUserBody() {

    constructor(firstname: String?, lastname: String?, email: String?, password: String?, gender: String?) : this() {
        this.firstname = firstname
        this.lastname = lastname
        this.email = email
        this.password = password
        this.gender = gender
    }

    var firstname: String? = null
    var lastname: String? = null
    @NotEmpty @Email var email: String? = null
    @NotEmpty var password: String? = null
    var gender: String? = null
}
